src/geant4/app/src/DagSolidMaterial.cc

```diff
@@ -93,39 +90,41 @@ std::map<int, G4Element*> get_g4elements(std::map<int, G4Isotope*> isotope_map)
 
 // using each g4element make the geant4 version of each pyne material
 std::map<std::string, G4Material*> get_g4materials(std::map<int, G4Element*> element_map,
-                                                   std::map<std::string, pyne::Material> material_library) {
+                                                   pyne::MaterialLibrary material_library) {
   std::map<std::string, G4Material*> material_map;
-  std::map<std::string, pyne::Material>::iterator it;
+
+  pyne::mat_map pyne_mat_map = material_library.get_mat_library();
+  pyne::mat_map::iterator it;
   pyne::comp_iter mat_it;
 
   std::string name;
   int ncomponents;
 
   // loop over the PyNE material library instanciating as needed
-  for (it = material_library.begin() ; it != material_library.end() ; ++it) {
+  for (it = pyne_mat_map.begin() ; it != pyne_mat_map.end() ; ++it) {
     // get the material object
-    pyne::Material mat = it->second;
+    std::shared_ptr<pyne::Material> mat = it->second;
```

Comment:
Does this need to be a `shared_ptr`?  It seems most of the other changes are a consequence of this one.  Maybe keep that separate from this PR?